src/main/java/it/eng/idsa/dataapp/handler/ArtifactMessageHandler.java

```diff
@@ -53,22 +65,13 @@ public Map<String, Object> handleMessage(Message message, Object payload) {
 
 			logger.debug("Handling message with requestedElement:" + arm.getRequestedArtifact());
 
-			// Check if requested artifact exist in self description
-			if (selfDescriptionService.artifactRequestedElementExist(arm,
-					selfDescriptionService.getSelfDescription(message))) {
-				if (isBigPayload(arm.getRequestedArtifact().toString())) {
-					payload = encodePayload == true ? encodePayload(BigPayload.BIG_PAYLOAD.getBytes())
-							: BigPayload.BIG_PAYLOAD;
-				} else {
-					payload = encodePayload == true ? encodePayload(createResponsePayload().getBytes())
-							: createResponsePayload();
-				}
+			if (((Boolean) threadService.getThreadLocalValue("wss") != null
```

Comment:
Can we use check like in this post?
https://community.sonarsource.com/t/new-rule-suggestion-boolean-objects-should-be-used-in-a-null-safe-way-inside-of-expressions/2024?gads_campaign=SQ-Class03-Generic&gads_ad_group=DSA&gads_keyword=&gclid=EAIaIQobChMI1uOln6Pq_QIVmOJ3Ch28ywvoEAAYASAAEgKQhPD_BwE

This one should also cover null check

Reply:
Great suggestion - done!